packages/api/src/apps/api/app.ts

```diff
@@ -97,25 +98,46 @@ async function run(env: ProcessEnv) {
   // expose calls through express
   await Express({ port: Number(env.EXPRESS_PORT) }, actions);
 
-  // main update loop, update every block
-  provider.on("block", (blockNumber: number) => {
+  // break all state updates by block events into a cleaner function
+  async function updateByBlock(blockNumber: number) {
+    await services.blocks.handleNewBlock(blockNumber);
     // dont do update if this number or blocks hasnt passed
-    services.blocks.handleNewBlock(blockNumber).catch(console.error);
     if (blockNumber - appState.lastBlockUpdate >= updateBlocks) {
-      services.registry(appState.lastBlock, blockNumber).catch(console.error);
-      services.emps(appState.lastBlock, blockNumber).catch(console.error);
+      // update everyting
+      await services.registry(appState.lastBlock, blockNumber);
+      await services.emps(appState.lastBlock, blockNumber);
+      await services.erc20s.update();
+      await services.empStats.update();
+
       appState.lastBlockUpdate = blockNumber;
-      services.erc20s.update().catch(console.error);
-      services.empStats.update().catch(console.error);
     }
     appState.lastBlock = blockNumber;
-    services.blocks.cleanBlocks(oldestBlock).catch(console.error);
+    await services.blocks.cleanBlocks(oldestBlock);
+  }
+
+  // main update loop, update every block
+  provider.on("block", (blockNumber: number) => {
+    // this timeid/time stuff just logs how long the update takes
+    const timeid = uniqueId("App State Updated ");
+    console.time(timeid);
```

Comment:
was this left over?

Reply:
i kind of left this in as some logging to profile the app as it runs, its definitely not necessary and can remove it

Reply:
removed these to avoid confusion